base/strings/string.jl

```diff
@@ -27,7 +27,10 @@ function Base.showerror(io::IO, exc::StringIndexError)
     end
 end
 
-const ByteArray = Union{CodeUnits{UInt8,String}, Vector{UInt8},Vector{Int8}, FastContiguousSubArray{UInt8,1,CodeUnits{UInt8,String}}, FastContiguousSubArray{UInt8,1,Vector{UInt8}}, FastContiguousSubArray{Int8,1,Vector{Int8}}}
+const ByteArray = Union{
+    DenseVector{<:Union{UInt8, Int8}},
```

Comment:
This is not a byte array, but rather a supertype of some byte arrays and some non-byte arrays

Reply:
I don't understand. The type there is a subtype of `AbstractArray{<:Union{UInt8, Int8}}`. Surely that's an array of bytes?
The name is a little poor. It's supposed to be a memory-backed dense bytearray.
Are you objecting to the name, or to the definition of the type? If so, what would be better?

Reply:
I believe it's an objection to the name. (codeunits isn't an array). perhaps densebytes?

Reply:
Codeunits is absolutely an array:
```julia
julia> Base.CodeUnits <: DenseArray
true
```

Reply:
Only a bit less than half of the concrete subtypes of it are arrays of bytes, since `DenseVector{<:Union{UInt8, Int8}} >: DenseVector{UInt8} + DenseVector{Int8} + DenseVector{Union{UInt8, Int8}} + DenseVector{Union{}}`

Reply:
FWIW, the fallback method interface (io.jl) calls this query `!isbitstype(T) || _checkcontiguous(Bool, A)` for `A::StridedArray{T}`, so we might want to be consistent and use the same query in all the subtypes here